Sources/NIOCore/AsyncChannel/AsyncChannel.swift

```diff
@@ -284,6 +284,7 @@ public struct NIOAsyncChannel<Inbound: Sendable, Outbound: Sendable>: Sendable {
     /// - Important: After this method returned the underlying ``Channel`` will be closed.
     ///
     /// - Parameter body: A closure that gets scoped access to the inbound and outbound.
+    @_disfavoredOverload
```

Comment:
I think we should also deprecate those methods since in theory those are not concurrency safe and we want users to adopt a Swift 6 compiler to make those methods safe again.

Reply:
I don't think it's appropriate to use deprecation to encourage users to update _compiler_ version. Deprecation warnings really need to be something that can be resolved without a toolchain change.